tree_test.go

```diff
@@ -135,13 +135,16 @@ func TestTreeWildcard(t *testing.T) {
 
 	routes := [...]string{
 		"/",
-		"/cmd/:tool/:sub",
 		"/cmd/:tool/",
+		"/cmd/:tool/:sub",
```

Comment:
Thanks for all the updates, I really appreciate it.

I think we could use another test to check edge case to check:


if you have routes like this set up:
* `/cmd/whoami/foo`
* `/cmd/:tool/`
* `/cmd/:tool/:subtool`

What happens if someone hits the route:
* `/cmd/whoami/bar`?

It should cause a 404 and not hit `/cmd/:tool/:subtool`.

Basically, whenever the tree is exhausted or a the end of a path segment is found, the backtrack should be invalidated. Once you hit `/cmd/whoami/`, it shouldn't be possible to fall back to `/cmd/tool/`.

Reply:
This is the problem.

For example, `/cmd/who/` will not hit `/cmd/:tool/` because it has the same prefix as `/cmd/whoami/foo`, but it should match.

Reply:
Know what you mean. I will think about this situation.

Reply:
But what still let me confused is: with the above routes, whether `/cmd/whoami/bar` hit or not, seems they are meaningful?

According to my understanding and my actual use, if I don’t have an exact route definition like `/cmd/whoami/foo`, then `/cmd/whoami/bar` needs to be matched to the wildcard route.